src/Microsoft.HttpRepl/Preferences/UserFolderPreferences.cs

```diff
@@ -27,15 +46,80 @@ public string PreferencesFilePath
                 }
                 return _prefsFilePath;
             }
-        }         
+        }
 
-        public UserFolderPreferences(IFileSystem fileSystem, IUserProfileDirectoryProvider userProfileDirectoryProvider)
+        public UserFolderPreferences(IFileSystem fileSystem, IUserProfileDirectoryProvider userProfileDirectoryProvider, IDictionary<string, string> defaultPreferences)
         {
             _fileSystem = fileSystem;
             _userProfileDirectoryProvider = userProfileDirectoryProvider;
+            DefaultPreferences = SetupDefaults(defaultPreferences);
+        }
+
+        public AllowedColors GetColorValue(string preference, AllowedColors defaultValue = AllowedColors.None)
+        {
+            if (!Preferences.TryGetValue(preference, out string preferenceValueString) || !Enum.TryParse(preferenceValueString, true, out AllowedColors result))
```

Comment:
There's a lot of branches added to this class (each of these ifs has 3 branches).  How's code coverage looking?

Reply:
Note that those three functions were basically just moved from the WellKnownPreferences.cs file. That said, I had a bunch of tests in the works for them today and they just went in with c430a00.